src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets

```diff
@@ -169,7 +163,23 @@ Copyright (c) .NET Foundation. All rights reserved.
     <Message Importance="High" Text="$(MSBuildProjectName) (Blazor output) -&gt; $(TargetDir)wwwroot" Condition="'$(CopyBuildOutputToOutputDirectory)' == 'true' and '$(SkipCopyBuildProduct)'!='true'" />
   </Target>
 
-  <!-- Publish starts here -->
+  <Target
+    Name="_CollectAssetsToCompressForPublish"
+    DependsOnTargets="ProcessPublishFilesForWasm;ComputeWasmExtensions;_AddPublishWasmBootJsonToStaticWebAssets">
+    <ItemGroup>
+      <AssetToCompress
+        Include="@(StaticWebAsset)"
+        Condition="'%(AssetKind)' != 'Build' and ('%(StaticWebAsset.AssetTraitName)' == 'WasmResource' or '%(StaticWebAsset.AssetTraitName)' == 'Culture')">
+        <ConfigurationName>BlazorPublishGzip</ConfigurationName>
+      </AssetToCompress>
+
+      <AssetToCompress
+        Include="@(StaticWebAsset)"
+        Condition="'%(AssetKind)' != 'Build' and ('%(StaticWebAsset.AssetTraitName)' == 'WasmResource' or '%(StaticWebAsset.AssetTraitName)' == 'Culture')">
+        <ConfigurationName>BlazorPublishBrotli</ConfigurationName>
+      </AssetToCompress>
+    </ItemGroup>
+  </Target>
```

Comment:
Could we express this in terms of file patterns?

Reply:
Maybe. I tried this but found it caused issues in some cases. See https://github.com/dotnet/sdk/pull/31559#discussion_r1154881408

Reply:
Were you running into ordering problems?